File: SFA-API/Domain/Store.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Store
    {
        public int Id { get; set; }
        public int StoreManagerId { get; set; }
        public int RouteId { get; set; }
        public string StoreCode { get; set; }
        public string Name { get; set; }
        public string Comment { get; set; }
        public DateTime CreatedDate { get; set; }
        public string CreatedBy { get; set; }
        public virtual ICollection<StoreAddress> StoreAddresses { get; set; }
        public virtual ICollection<StoreEmail> StoreEmails { get; set; }
        public virtual ICollection<StoreContact> StoreContacts { get; set; }
        public virtual UserDetail UserDetail { get; set; }
        public virtual Route Route { get; set; }
        public virtual ICollection<StoreItemBatch> StoreItemBatches { get; set; }
    }
}

File: SFA-Client/src/app/_models/store.ts
import { District } from './district';

export class Store {
  id: number;
  routeId: number;
  routeName: string;
  storeManagerId: number;
  storeManagerName: string;
  storeCode: string;
  name: string;
  comment: string;
  displayAddress: string;
  createdDate: Date;
  createdBy: string;
  loginEmail: string;
  storeAddresses: StoreAddress[];
  storeEmails: StoreEmail[];
  storeContacts: StoreContact[];

  constructor(store) {
    this.id = store.id;
    this.routeId = store.routeId;
    this.routeName = store.routeName;
    this.storeManagerId = store.storeManagerId;
    this.storeManagerName = store.storeManagerName;
    this.storeCode = store.storeCode;
    this.name = store.name;
    this.comment = store.comment;
    this.displayAddress = store.displayAddress;
    this.createdBy = store.createdBy;
    this.createdDate = store.createdDate;
    this.loginEmail = store.loginEmail;
    this.storeAddresses = store.storeAddresses;
    this.storeEmails = store.storeEmails;
    this.storeContacts = store.storeContacts;
  }
}

export interface StoreAddress {
  id: number;
  storeId: number;
  provinceId: number;
  districtId: number;
  address1: string;
  address2: string;
  address3: string;
  locationLatitude: number;
  locationLongitude: number;
}

export interface StoreEmail {
  id: number;
  email: string;
  storeId: number;
}

export interface StoreContact {
  id: number;
  contactNo: string;
  storeId: number;
}
